v2/proto/wallet.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto3";
+
+package pb;
+
+// TODO - timestamps need not be uint64 right?
+
+message Wallet {
+  repeated Account accounts = 1; // Accounts associated with this wallet
+  string name = 2; // Human readable name for this wallet
+  TimestampedPreferences preferences = 3;
+  uint32 version = 4;  // Wallet spec version
+}
+
+message Account {
+  message AddressGenerator { // Meta-manager for both singular or deterministically generated addresses
+    message AddressManager { // Manager for deterministically generated addresses
+      // TODO - it seems like chain can be as high as (1 << 32) - 1. I think that means that uint32 should cover `gap`?
+      uint32 gap = 1; // Maximum allowed consecutive generated addresses with no transactions
+      uint32 maximum_uses_per_address = 2 [json_name="maximum_uses_per_address"]; // Maximum number of uses for each generated address
+    }
+    string name = 1; // type of address generator: "deterministic-chain" or "single-address"
```

Comment:
@eukreign can this name field contain any name or is it choosing from a fixed set of options?

Reply:
just the two listed in the comment

Reply:
@orblivion can you make these `oneOf`s plz

Reply:
I thought `oneof` was for fields (and thus types), not values. The only thing I saw for values was an enum, and that's only integers.

Reply:
Now, if this is valuable enough that schema changes are on the table, an enum may be the way to go.

```
enum AddressManagerType {
  single_address = 0;
  deterministic_chain = 1;
}
```

Reply:
if enum doesn't do strings, there are two options. one is to leave it as a string. the other is to have it as an enum but then convert that to a string on save/load.

i think leaving it as a string is fine here

Reply:
If we need to do custom conversion code for the preferences json blob, I could do that here too.